homeassistant/components/proximity.py

```diff
@@ -0,0 +1,325 @@
+            return
+
+        # stop if we cannot calculate the direction of travel (i.e. we don't
+        # have a previous state and a current LAT and LONG)
+        if old_state is None or 'latitude' not in old_state.attributes:
+            proximity.dist_to = round(distances_to_zone[entity])
+            proximity.dir_of_travel = 'unknown'
+            proximity.nearest = entity_name
+            proximity.update_ha_state()
+            _LOGGER.debug('%s: update entity: distance=%s: direction='
+                          'unknown: device=%s', entity_id,
+                          dist_to_zone, entity_name)
+            _LOGGER.info('%s: cannot determine direction of travel: old '
+                         'and/or new LAT or LONG are missing', entity_name)
+            return
+
+        # reset the variables
+        distance_travelled = 0
+
+        # calculate the distance travelled
+        old_distance = distance(proximity_latitude, proximity_longitude,
+                                old_state.attributes['latitude'],
+                                old_state.attributes['longitude'])
+        _LOGGER.debug('%s: old distance=%s', entity_name, old_distance)
+        new_distance = distance(proximity_latitude, proximity_longitude,
+                                new_state.attributes['latitude'],
+                                new_state.attributes['longitude'])
+        _LOGGER.debug('%s: new distance=%s', entity_name, new_distance)
+        distance_travelled = round(new_distance - old_distance, 1)
+
+        # check for tolerance
+        if distance_travelled < tolerance * -1:
+            direction_of_travel = 'towards'
+            _LOGGER.info('%s: travelled=%s: moving=%s', entity_name,
+                         distance_travelled, direction_of_travel)
+        elif distance_travelled > tolerance:
+            direction_of_travel = 'away_from'
+            _LOGGER.info('%s: travelled=%s: moving=%s', entity_name,
+                         distance_travelled, direction_of_travel)
+        else:
+            direction_of_travel = 'stationary'
+            _LOGGER.info('%s: distance travelled too small: %s',
+                         entity_name, distance_travelled)
+
+        # update the proximity entity
+        proximity.dist_to = round(dist_to_zone)
+        proximity.dir_of_travel = direction_of_travel
+        proximity.nearest = entity_name
+        proximity.update_ha_state()
+        _LOGGER.debug('%s update entity: distance=%s: direction=%s: '
+                      'device=%s', entity_id, round(dist_to_zone),
+                      direction_of_travel, entity_name)
+
+        _LOGGER.info('%s: proximity calculation complete', entity_name)
+
+    # main command to monitor proximity of devices
+    track_state_change(hass, proximity_devices,
+                       check_proximity_state_change)
+
+    # Tells the bootstrapper that the component was successfully initialized
+    return True
+
+
+class Proximity(Entity):
```

Comment:
Since your state is expressed in kilometers, you should add the property `unit_of_measurement`